File: Projects/MultiFileReader/modules/yml_parser.py
#
# Nimish Nayak
# 08-06-2017
#

#
# YAML Parser 
#

# required libraries
import logging
import yaml
from parser import Parser 

class YmlParser(Parser):

    # initialize the instance variables
    def __init__(self, input_file):
    	# call the base class constructor
    	Parser.__init__(self, input_file)
        logging.debug("Yml parser initialized")

    # parse the file using a xml parser 
    def parse_file(self):
		with open(self.input_file, "r") as f:
			data = yaml.load(f)
			if data and "users" in data:
				self.data = data["users"]

File: modules/yml_parser.php
<?php

#
# Nimish Nayak
# 10-06-2017
#

#
# YAML Parser 
#

# required libraries
require_once("./modules/parser.php");
require_once("./modules/yaml/lib/sfYaml.php");
    
class YmlParser extends Parser {

    # initialize the instance variables
    function __construct($inputFile){
        
        # call the base class constructor
        parent::__construct($inputFile);
        echo("Yml parser initialized\n");
    }

    # parse the file using a xml parser 
    public function parseFile(){

        try {
            
            $this->doc = sfYaml::load($this->inputFile);

            #var_dump($this->doc);

            if ($this->doc['users']) {

                # we have yml data!
                # parse and create an associative array from it
                $this->data = array();

                foreach ($this->doc['users'] as $user) {

                    $dict = array('name' => $user['name'], 'active' => Utils::strToBool($user['active']), 'value' => (int)$user['value']);
                    array_push($this->data, $dict);

                }
            }
        }
        catch (Exception $e) {
            echo($e->getMessage());
        }
    }

}

?>